beacon_chain/genesis/src/beacon_block.rs

```diff
@@ -47,4 +47,59 @@ mod tests {
         // This only checks that the function runs without panic.
         genesis_beacon_block(state_root, &spec);
     }
+
+    // Tests items that are 0 or zero_hash
+    #[test]
+    fn test_zero_items() {
+        let spec = ChainSpec::foundation();
+
+        // Note: state_root will not be available without a state (test in beacon_state)
+        let state_root = Hash256::zero();
+
+        let genesis_block = genesis_beacon_block(state_root, &spec);
+
+        assert!(genesis_block.slot == 0);
+        assert!(genesis_block.parent_root.is_zero());
+        assert!(genesis_block.randao_reveal.is_zero());
+        assert!(genesis_block.candidate_pow_receipt_root.is_zero()); // aka deposit_root
+    }
+
+    // Tests the BeaconBlockBody inside BeaconBlock
+    #[test]
+    fn test_beacon_body() {
+        let spec = ChainSpec::foundation();
+
+        // Note: state_root will not be available without a state (test in beacon_state)
+        let state_root = Hash256::zero();
+
+        let genesis_block = genesis_beacon_block(state_root, &spec);
+
+        // Custody items are not being implemented until phase 1 so tests to be added later
+
+        assert!(genesis_block.body.proposer_slashings.is_empty());
+        assert!(genesis_block.body.casper_slashings.is_empty());
+        assert!(genesis_block.body.attestations.is_empty());
+        assert!(genesis_block.body.deposits.is_empty());
+        assert!(genesis_block.body.exits.is_empty());
+    }
+
+    #[test]
+    fn test_signature() {
+        let spec = ChainSpec::foundation();
+
+        // Note: state_root will not be available without a state (test in beacon_state)
+        let state_root = Hash256::zero();
+
+        let genesis_block = genesis_beacon_block(state_root, &spec);
+
+        // Signature should consist of [bytes48(0), bytes48(0)]
+        // Note this is implemented using Apache Milagro BLS which requires one extra byte -> 97bytes
+        let raw_sig = genesis_block.signature.as_raw();
+        let raw_sig_bytes = raw_sig.as_bytes();
+
+        assert!(raw_sig_bytes.len() == 97);
```

Comment:
I don't think we need to test the byte length here.

In production, we use ssz to go to bytes (e.g., `genesis_block.signature.ssz_encode()`). It's the concern of SSZ to ensure it enc/decodes as per the spec, not the genesis functions.